### ISSUE TITLE
How to evaluate your code on my own database?

### ISSUE THREAD

**Issue by FesianXu** [jz462/Large-Scale-VRD.pytorch#10]
Thanks for your wonderful work, It really inspires me a lot.
And I wanna evaluate your code on my own database to do my further research, Could you please give me some advices on how to modify your code? Your code seems complex and hard to modify. Thanks for any suggestions.

Best Regard.

**Comment by clungzta** [comment]
Agreed! An excellent project, it'd be significantly more approachable if a simple helper function existed to run inference on a pre-trained model given a provided image (path or ndarray).

**Comment by FesianXu** [comment]
> Agreed! An excellent project, it'd be significantly more approachable if a simple helper function existed to run inference on a pre-trained model given a provided image (path or ndarray).

There is not existing example toy-data so we need to download the VRD or VG database at first. actually, according to my struggle experience on this project code, just following the readme is ok though there are still some troubles you will meet in the configuration, see :https://github.com/jz462/Large-Scale-VRD.pytorch/issues/11
But the author emailed me and said that he had re-trained the model so I am still trying his new model in pytorch version code.

btw, I have read the caffe2 version code, link: https://github.com/facebookresearch/Large-Scale-VRD
In that version, just following the readme and download the database and the weights files, then your can reproduce the result reported in the paper. I am now working on the caffe2 version code.

**Comment by mrfarazi** [comment]
Hello @FesianXu @clungzta ,
Did you find a way to run inference on a custom dataset? I can see that in test_net_rel.py you need to provide the GT bounding boxed (at least for VG dataset). Is there a way to by pass that?

**Comment by FesianXu** [comment]
@mrfarazi 
finally i turned to caffe2 version. caffe2 version VRD code seems easier to understand and reproduce.

**Comment by mrfarazi** [comment]
Migrating to Caffe2 might not be possible for my project.
I just need to run inference on a custom image outside the val dataset and extract  <subject> - <predicate> - <object> triplets.

**Comment by achireistefan** [comment]
@mrfarazi 
I just managed to build the project, with a little struggle but now it works. 
Did u managed to ajust the code in the caffe2 implementation for custom test dataset?

**Comment by mrfarazi** [comment]
@achireistefan no, i didn't switch to Caffe2. I was using PyTorch

**Comment by ffcarina** [comment]
@mrfarazi  Hi, Did you solve your problem? Do you know how to make inference on a custom image?
